### DISCUSSION TITLE
View 'local photos' from within Immich

### DISCUSSION THREAD

**Discussion by JordyEGNL** [immich-app/immich#2266] [Feature Request]
Make it possible to view local photos, that are not backed up by Immich.

Now you need to open your phones Gallery app for that. 
Google Photos also has this feature under, Library > Photos on device > view all.

From here you can also enable backups for a particular folder.

Would be cool to also have this feature in Immich!

**Comment by alextran1502** [comment]
It is currently possible, however Immich will only show local photos  in the album that you select to be uploaded

**Reply by jrasm91** [reply]
I'll just add that it is possible, but the way to accomplish it is not obvious. If you go to the backup screen, then album selection, you can click the picture icon in the right to view the photos in the album. I like the Google photos navigation better myself.

**Reply by pec0ra** [reply]
I often encounter the issue for the following workflow:

1. Take a screenshot
2. Find it in gallery (and sometimes edit it)
3. Share it

I generally don't want to backup screenshots since they are mostly temporary images that I want to share. I can therefore not find them in the "Albums on Device" section of immich.

Any particular reason to not show all albums instead of only the ones selected for upload? It could even be a separate section in the "Library" page.

**Reply by devinpatel** [reply]
I'm currently running into an issue where I want to auto backup all my camara roll items and only selectively upload my WhatsApp images and videos, but this isn't possible since the the albums that you select for backup on the backup screen is directly tied to what albums show up in the Timeline and `Library > On this devices` sections of the Android app.

There are some items from my "WhatsApp Images" and "WhatsApp Videos" albums that I don't want to backup so to be able to selectively back them up I have to select those albums in the backup screen so that they will show up in the Timeline, then this allows me to selectively upload only the items I want. The issue with this is that now I can't easily backup/auto-backup entire albums any more, like my Camera roll, since the WhatsApp albums are currently selected on the backup screen.

It would be nice to have separate configurations for what albums show up in the timeline and what albums you want backed up.

**Comment by Pheggas** [comment]
What do you mean @JordyEGNL ? It is already possible on mobile app.

**Reply by JordyEGNL** [reply]
The Google Photos app shows way more folders, such as WhatsApp Images, WhatsApp Videos, Screenshots, Infinity and Telegram.

Btw I am talking about the Library screen and not the Backup Albums screen.

**Reply by JordyEGNL** [reply]
As you can see in the screenshots below, Google Photos shows all the folders with images in them, and Immich only shows folders within the folder Pictures and DCIM.

It is very common for me to search a picture in the Whatsapp Images folder, These are located in the app folder in /Android/media/com.whatsapp/WhatsApp/Media.

<img src="https://github-production-user-asset-6210df.s3.amazonaws.com/65036298/243642305-0dc9d593-4bf6-446b-98cc-fa265ac22de8.jpg" width=30% height=30%> <img src="https://github-production-user-asset-6210df.s3.amazonaws.com/65036298/243642321-d858f2f8-5177-4c2c-b188-d23452117dc9.jpg" width=30% height=30%>

**Reply by Pheggas** [reply]
Right so Immich has lack of browsable local folders?

**Reply by JordyEGNL** [reply]
> Right so Immich has lack of browsable local folders?

Yeah, it doesn't seem to search every folder with images in them

**Comment by matichewer** [comment]
@JordyEGNL  Do you know anything new about this? I'm starting to use Immich and I can't uninstall my old gallery because I need it to explore the local image folders that I don't have in backup

**Reply by mmomjian** [reply]
Immich does not support this feature at this time, thanks for understanding.

**Comment by xd003** [comment]
Is there any update on this ? this is a much needed feature. One would not want to upload everything. Having to use a separate gallery app just to view local photos is quite inconvenient

**Reply by pec0ra** [reply]
Since v1.107.0 which has been released yesterday, all local photos are displayed in the timeline. Including those that are not in an directory selected to be backed up.

**Reply by pec0ra** [reply]
Sorry for the misinformation, it seems like the change has been reverted in v1.107.1. They are working on it though!

**Reply by ZoTay** [reply]
any idea why it got reverted?

**Comment by andreigae** [comment]
Is there any news on this? It is an extremely important feature and without it I have to use another application to manage the local files. So it's very difficult to stop using the other commercial services as I don't want to have 2 or 3 apps for a basic function which is to view local photos without having to back them all up But only the folders of interest like camera etc.

**Comment by bavarianbidi** [comment]
My immich server instance was not available due to an issue (which was unrelated to immich itself). Unfortunately i was on vacation and with that AFK for more than a week to fix the server configuration in my home lab.

That circumstance brought the downside, that on my mobile phone (Android), i was not able to use immich as "default image viewer" to see the recent pictures i've taken with the camera.
I had to fix the server first to be able to see the images in immich mobile app.

This leads to two different questions:
* if a folder is configured for "auto-upload" and the images aren't uploaded yet (for different reasons: e.g. immich server is down) should immich mobile app display not uploaded images?
* will immich mobile app became an media-library app for all media (independent if images are managed to get uploaded to an immich server or not - xref: https://github.com/immich-app/immich/discussions/2266#discussioncomment-9027420)

**Comment by marsteff** [comment]
any news about that? The last reply is a little old.

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by deam0n** [comment]
This would be a pretty nice addition to finally allow me to degoogle from google photos. The lack of this feature in immich, makes me open google photos every time I download some image from an app, or take a screenshot...

**Comment by mpvennik** [comment]
Immich is a wonderful piece of software but in this regard it still falls short. (Basic) Local photo management  is imho a must. Let me explain with an example. I've just quit WhatsApp but kept the image folder. A lot of images are jokes, etc and can be deleted but there are also images I want to keep.  Because Immich doesn't  do locals I need another app  to select and sort this out.

**Comment by popy2k14** [comment]
As stated by others, I second this. 
My Google pixel has "Google Photos" as gallery app.
To complete the switch to mich, it would be nice if I can completely ditch Google Fotos for my family members and just use immich.

**Comment by funkemunky** [comment]
Are there any blockers on this that need addressing? I was unable to find any related Issues, PRs, or action items within the Projects tab. If the maintainer team is interested in adding this but needs some help, please reply and let me know where work needs to get started/improved. Eager to help where I can.

**Reply by alextran1502** [reply]
This is addressed in #19370 as well. Implemented, just need beta testing to roll out

**Reply by aaronjamt** [reply]
> This is addressed in #19370 as well. Implemented, just need beta testing to roll out

Can we help beta test? Since it's only mobile-app changes I'd be happy to run that on my phone to help test its stability.

**Reply by antoniovj1** [reply]
If I can help testing it count on me, it's the only piece I needed to uninstall Google Photos🎉

**Reply by JordyEGNL** [reply]
> If I can help testing it count on me, it's the only piece I needed to uninstall Google Photos🎉

This is already implemented :)

Open your library and click on 'on this device'

**Reply by antoniovj1** [reply]
😮 Thanks!

**Comment by stimpy2k** [comment]
While I understand it's possible to navigate to Library, tap on On This Device, and then tap on Camera (for example), I think what people are asking for (me along with them) is the ability to see these local folders in the main Photos section of the app _without_ also having to upload them. Like others have mentioned, I find it cumbersome to go through a 3 step process just to see a picture I just took.

As others have mentioned, Google Photos handles this quite well. I literally only still use Google Photos to work with my local media. Would be great to fully disconnect from that and only use Immich (which is great).

**Reply by alextran1502** [reply]
You can select albums for backup which will then show you the photos in the albums on the timeline, but you don't need to toggle the "Enable backup" button

**Reply by antoniovj1** [reply]
Some of us don't have this enabled because we cannot backup to an external library. I think for some of us, when that can be done, both "issues" will be solved 👌

**Reply by gaeandrei** [reply]
While I totally understand that it works the way @alextran1502 is saying, I think it would be a great feature for the future to be able to automatically back up photos from my camera (or specific folders), but also see WhatsApp photos in the timeline without backing them up. 

That way, I can see all the media I received without having to go to Library > Files on this device.

The current system is pretty close, but it lacks one specific feature: disabling automatic backup for a specific folders that are selected to be backed up just so they are shown in the timeline.
Right now, the automatic backup works like 'backup everything or nothing' for files shown in the timeline.

It is great to automatically back up the photos I take from my camera, but I don't want to back up all the photos that my friends send to me. However, I still want to see them on my timeline.

Since there is currently no way to select which folders to show in the timeline without backing them up while maintaining the automatic backup, I have had to fully disable automatic backup for everything.

I think lots of us are doing the same, losing a very important feature that saves our photos if our devices break or get stolen.

**Reply by raleik-pl** [reply]
This is a very same thing I am looking for, and Immich would be definitelly 100% satisfying my needs if that was possible (now it's "just" 99% ;)
Exactly the same usecase I have - I want my camera photos to be backed-up automatically (and so they are), and I want my WhatsApp images I reveice to show up alltogether with my photos, but without being backed-up automatically (lot's of trash/gifs/etc I don't want stored).
This way, from a single view, I could select photos I took and the photos I received, to e.g. share them to Image Toolbox (to make a collage), or to send them all to someone else, or to manually upload the ones that aren't auto-backed-up.
That's really a needed feature!
